web/src/typing.js

```diff
@@ -56,6 +75,19 @@ function send_typing_notification_based_on_message_type(to, operation) {
     }
 }
 
+function send_typing_notifications_for_message_edit(message_id, operation) {
+    if (get_message_edit_typing_notification_policy()) {
```

Comment:
This function is poorly named; the name in no way indicates the type is boolean rather than an enum that is the value of the policy, which one would not want to use in this sort of conditional. Call it something like `message_edit_typing_notifications_enabled()`.

Reply:
Oh and it should accept the message object or ID, and check if that message is a stream/direct message to determine which policy to apply. Looking at `narrow_state` is incorrect -- views like "all messages" can contain both direct and stream messages, and we need to check the policy for the actual message in question.

Reply:
I guess in that case, `message_store` should do the work right...?